go/libraries/doltcore/sqle/dfunctions/dolt_tag.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+)
+
+const DoltTagFuncName = "dolt_tag"
+
+// DoltTagFunc runs a `dolt commit` in the SQL context, committing staged changes to head.
+// Deprecated: please use the version in the dprocedures package
+type DoltTagFunc struct {
+	expression.NaryExpression
+}
+
+// NewDoltTagFunc creates a new DoltTagFunc expression whose children represent the args passed in DOLT_COMMIT.
+// Deprecated: please use the version in the dprocedures package
+func NewDoltTagFunc(args ...sql.Expression) (sql.Expression, error) {
+	return &DoltTagFunc{expression.NaryExpression{ChildExpressions: args}}, nil
+}
+
+func (d DoltTagFunc) String() string {
+	childrenStrings := make([]string, len(d.Children()))
+
+	for i, child := range d.Children() {
+		childrenStrings[i] = child.String()
+	}
+
+	return fmt.Sprintf("DOLT_BRANCH(%s)", strings.Join(childrenStrings, ","))
+}
+
+func (d DoltTagFunc) Type() sql.Type {
+	return sql.Int8
+}
+
+func (d DoltTagFunc) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	return NewDoltBranchFunc(children...)
+}
+
+func (d DoltTagFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	args, err := getDoltArgs(ctx, row, d.Children())
+	if err != nil {
+		return 1, err
+	}
+	return DoDoltTag(ctx, args)
+}
+
+func DoDoltTag(ctx *sql.Context, args []string) (int, error) {
```

Comment:
Add some godocs for any exported functions and consider it for any complex private functions.